sql/core/src/test/scala/org/apache/spark/sql/DataFrameWriterV2Suite.scala

```diff
@@ -531,6 +532,15 @@ class DataFrameWriterV2Suite extends QueryTest with SharedSparkSession with Befo
     assert(table.properties === (Map("provider" -> "foo") ++ defaultOwnership).asJava)
   }
 
+  test("SPARK-39543 writeOption should be passed to storage properties when fallback to v1") {
+    spark.range(10).writeTo("table_name").option("compression", "zstd").using("parquet").create()
```

Comment:
can we test with `InMemoryV1Provider`? We may migrate the file source to v2 completely in the future and then this test won't test the v1 fallback anymore.

Reply:
thanks